nixos/modules/services/web-apps/cryptpad.nix

```diff
@@ -0,0 +1,447 @@
+    # UI content is loaded via the `sandbox_domain`.
+    # "Content Security Policy" headers prevent content loaded via the sandbox
+    # from accessing privileged information.
+    # These variables must be different to take advantage of CryptPad's sandboxing techniques.
+    # In the event of an XSS vulnerability in CryptPad's front-end code
+    # this will limit the amount of information accessible to attackers.
+    set $main_domain "${nginx_main_domain}";
+    set $sandbox_domain "${nginx_sandbox_domain}";
+
+    # By default CryptPad forbids remote domains from embedding CryptPad documents in iframes.
+    # The sandbox domain must always be permitted in order for the platform to function.
+    # If you wish to enable remote embedding you may change the value below to "*"
+    # as per the commented value.
+    set $allowed_origins "https://''${sandbox_domain}";
+    #set $allowed_origins "*";
+
+    # CryptPad's dynamic content (websocket traffic and encrypted blobs)
+    # can be served over separate domains. Using dedicated domains (or subdomains)
+    # for these purposes allows you to move them to a separate machine at a later date
+    # if you find that a single machine cannot handle all of your users.
+    # If you don't use dedicated domains, this can be the same as $main_domain
+    # If you do, they can be added as exceptions to any rules which block connections to remote domains.
+    # You can find these variables referenced below in the relevant places
+    set $api_domain "${nginx_main_domain}";
+    set $files_domain "${nginx_main_domain}";
+
+    # Speeds things up a little bit when resuming a session
+    ssl_session_timeout 1d;
+    ssl_session_cache shared:MozSSL:10m;
+    ssl_session_tickets off;
+
+    ssl_protocols TLSv1.2 TLSv1.3;
+    ssl_ciphers ECDHE-ECDSA-AES128-GCM-SHA256:ECDHE-RSA-AES128-GCM-SHA256:ECDHE-ECDSA-AES256-GCM-SHA384:ECDHE-RSA-AES256-GCM-SHA384:ECDHE-ECDSA-CHACHA20-POLY1305:ECDHE-RSA-CHACHA20-POLY1305:DHE-RSA-AES128-GCM-SHA256:DHE-RSA-AES256-GCM-SHA384;
+    ssl_prefer_server_ciphers off;
+
+    # HSTS (ngx_http_headers_module is required) (63072000 seconds)
+    add_header Strict-Transport-Security "max-age=63072000; includeSubDomains" always;
+
+    # OCSP stapling
+    ssl_stapling on;
+    ssl_stapling_verify on;
+
+    add_header X-XSS-Protection "1; mode=block";
+    add_header X-Content-Type-Options nosniff;
+    add_header Access-Control-Allow-Origin "''${allowed_origins}";
+    add_header Access-Control-Allow-Credentials true;
+    # add_header X-Frame-Options "SAMEORIGIN";
+
+    # Opt out of Google's FLoC Network
+    add_header Permissions-Policy interest-cohort=();
+
+    # Enable SharedArrayBuffer in Firefox (for .xlsx export)
+    add_header Cross-Origin-Resource-Policy cross-origin;
+    add_header Cross-Origin-Embedder-Policy require-corp;
+
+    # Insert the path to your CryptPad repository root here
+    root /var/lib/cryptpad;
```

Comment:
This is a bit ugly and fragile -- for example enabling DynamicUsers makes the directory unreadable to nginx. Might be work-aroundable with BindReadOnlyPaths in the service instead... but it works for default config, which will do for now

Reply:
Let's do small steps: let's get this somewhere and let's do follow up to stabilize further hardening, etc.